src/modules/site-v2/base/views/tools/phenotype_database.py

```diff
@@ -163,19 +163,25 @@ def get_zhang_traits_json():
     response_data = []
   return jsonify(response_data)
 
-@phenotype_database_bp.route('/traits-list')
+@phenotype_database_bp.route('/traits-list', methods=['POST'])
 @cache.memoize(60*60)
 @compress.compressed()
 def get_traits_json():
   """
     Get traits data for non-bulk files in JSON format (include phenotype values)
   """
-  try:
-    traits_query = query_phenotype_metadata()
-    traits_json = [ trait.to_json_with_values() for trait in traits_query]
-  except Exception:
-    traits_json = []
-  return jsonify(traits_json)
+  trait_name = request.json.get('trait_name')
+
+  if trait_name:
+    try:
+      trait = get_trait(trait_name).to_json_with_values()
+      return jsonify(trait)
+
+    except Exception as ex:
+      msg = f'Failed to retrieve metadata for trait {trait_name}'
+      logger.error({msg: ex})
+      
+  return jsonify({ 'message': msg }), 404
```

Comment:
if we are returning an error, besides the `404` statusCode`, we should standardize sending a key `error` in the response object. E.g.: `{'error': msg} `